dind/Dockerfile

```diff
@@ -1,26 +1,35 @@
 ARG DOCKER_VERSION
 
-FROM docker:${DOCKER_VERSION}-dind-rootless
+FROM golang:alpine AS build
+WORKDIR /build
```

Comment:
It's advised to set the workdir to /go/src/${MODULE_NAME} to maintain compatibility with building without go modules.

Reply:
Could you point me to a scenario where we would need that?

Reply:
a572e3a

Reply:
Indeed, we can keep it hardcoded then

Reply:
Some packages still don't work very well with go modules and fail to build if the project is outside the GO_PATH. Kubernetes is a good example of a project where this happens. To avoid this problem in the future, is always a good idea to build the project inside GO_PATH.